File: Tema 5 - Arrrays/ejercicio7/archives/automaticamente/automaticamente.php
<?php
if (isset($_POST["miArray"])) {
  $mensaje = "";
  $numero = json_decode($_POST["miArray"]);
  $colorCambio1 = 'background-color: yellow; color:black';
  $colorCambio2 = 'background-color: orange; color:black';
  $aux = [];
  $aux2 = [];
  $i = 0;
  foreach ($numero as $indice => $elemento) {
    if (abs($elemento) % 2 == 0) {
      $aux[$i] = $elemento;
      $auxpar[$indice] = $elemento;
      $i += 1;
    }
  }
  foreach ($numero as $indice => $elemento) {
    if (abs($elemento) % 2 == 0) {
    } else {
      $aux[$i] = $elemento;
      $auximpar[$indice] = $elemento;
      $i += 1;
    }
  }
  sort($auxpar);
  sort($auximpar);
  $ordenado = array_merge($auxpar, $auximpar);

  $mensaje = "<table border='1' >";
  //SECTION - fila indice
  $mensaje .= "<tr>";
  $mensaje .= "<th >Índice</th>";

  foreach ($numero as $indice => $elemento) {
    $mensaje .= "<th> $indice </th>";
  }
  $mensaje .= "</tr>";
  //!SECTION fin fila indice

  //SECTION - inicio fila matriz
  $mensaje .= "<tr>";
  $mensaje .= "<td >Matriz</td>";

  foreach ($numero as $indice => $valorArray) {
    if ($valorArray % 2 == 0) {
      $mensaje .= "<td style='$colorCambio1'> $valorArray  </td>";
    } else {
      $mensaje .= "<td style='$colorCambio2'> $valorArray </td>";
    }
  }

  $mensaje .= "</tr>";
  //!SECTION fin fila matriz
  //SECTION - inicio fila aux
  $mensaje .= "<tr>";
  $mensaje .= "<td >Aux</td>";

  foreach ($aux as $indice => $valorArray2) {
    if ($valorArray2 % 2 == 0) {
      $mensaje .= "<td style='$colorCambio1'> $valorArray2 </td>";
    } else {
      $mensaje .= "<td style='$colorCambio2'> $valorArray2 </td>";
    }
  }

  $mensaje .= "</tr>";
  //!SECTION fin fila aux

  //SECTION - inicio fila ordenado
  $mensaje .= "<tr>";
  $mensaje .= "<td >ordenado</td>";
  foreach ($ordenado as $indice => $valorArray2) {
    if ($valorArray2 % 2 == 0) {
      $mensaje .= "<td style='$colorCambio1'> $valorArray2 </td>";
    } else {
      $mensaje .= "<td style='$colorCambio2'> $valorArray2 </td>";
    }
  }

  $mensaje .= "</tr>";
  //!SECTION fin fila ordenado

  $mensaje .= "</table>";

  echo $mensaje;
} else {
  $mensaje = "Debe llenar ambos formularios";
  echo $mensaje;
}

File: Tema 5 - Arrrays/ejercicio7/archives/automaticamente/automaticamente.py
"""
Este módulo muestra un MENSAJE incrementando el valor de i.
"""
import sys
import json

NUMERO = json.loads(sys.argv[1])
MENSAJE = ""
COLORCAMBIO1 = "background-color: yellow; color:black"
COLORCAMBIO2 = "background-color: orange; color:black"
aux = []
aux2 = []
auxpar = []
auximpar = []

for indice, elemento in enumerate(NUMERO):
    if abs(elemento) % 2 == 0:
        aux.append(elemento)
        auxpar.append(elemento)

for indice, elemento in enumerate(NUMERO):
    if abs(elemento) % 2 == 0:
        pass
    else:
        aux.append(elemento)
        auximpar.append(elemento)

auxpar.sort()
auximpar.sort()
ordenado = auxpar + auximpar
MENSAJE = "<table border='1' >"

# SECTION - fila indice
MENSAJE += "<tr>"
MENSAJE += "<th >Índice</th>"
for indice, _ in enumerate(NUMERO):
    MENSAJE += f"<th> {indice} </th>"
MENSAJE += "</tr>"
#!SECTION fin fila indice

# SECTION - inicio fila matriz
MENSAJE += "<tr>"
MENSAJE += "<td >Matriz</td>"
for _, elemento in enumerate(NUMERO):
    if elemento % 2 == 0:
        MENSAJE += f"<td style='{COLORCAMBIO1}'> {elemento}</td>"
    else:
        MENSAJE += f"<td style='{COLORCAMBIO2}'> {elemento}</td>"
MENSAJE += "</tr>"
#!SECTION fin fila matriz

# SECTION - inicio fila aux
MENSAJE += "<tr>"
MENSAJE += "<td >Aux</td>"
for indice, elemento in enumerate(aux):
    if elemento % 2 == 0:
        MENSAJE += f"<td style='{COLORCAMBIO1}'> {elemento} </td>"
    else:
        MENSAJE += f"<td style='{COLORCAMBIO2}'> {elemento} </td>"
MENSAJE += "</tr>"
#!SECTION fin fila aux

# SECTION - inicio fila ordenado
MENSAJE += "<tr>"
MENSAJE += "<td >ordenado</td>"
for indice, elemento in enumerate(ordenado):
    if elemento % 2 == 0:
        MENSAJE += f"<td style='{COLORCAMBIO1}'> {elemento} </td>"
    else:
        MENSAJE += f"<td style='{COLORCAMBIO2}'> {elemento} </td>"
MENSAJE += "</tr>"
#!SECTION fin fila ordenado
MENSAJE += "</table>"


MENSAJE += "El array tiene:<br>"
MENSAJE += f"{len(auxpar)}"
MENSAJE += " numeros pares y<br>" if (len(auxpar) != 1) else " numero par y<br>"
MENSAJE += f"{len(auximpar)}"
MENSAJE += " numeros impares " if (len(auximpar) != 1) else " numero impar "
print(MENSAJE)
